### PR TITLE
AdRiver Bid Adapter: add new bid adapter

### PR DESCRIPTION
**Pull request by Fridoom007**
* Added AdRiver Bid Adapter

## Type of change

- [x] Feature
- [x] New bidder adapter 

## Description of change
Add new bid adapter

- test parameters for validating bids
```
{
 bidder: 'adriver',
      params: {
          siteid: '216200',
          placementId: '55:test_placement',
          bidfloor: 1
          dealid: 'dealidTest'
      }
}
```
- contact email of the adapter’s maintainer: aekhin@gramant.ru
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
https://github.com/prebid/prebid.github.io/pull/2820

### CODE REVIEW

**Comment by patmmccann** [comment]
Please set meta.advertiserDomains

**Review by jsnellbaker — CHANGES_REQUESTED** [review]
Hi @Fridoom007 

Thanks for submitting this bidder.  When you have the chance, I left some comments in-line below.  In addition, when I tried to use the test params in the PR description, I was receiving a 204 no content response from the adserver call.  

Could you take a look too see why the test bid wasn't returning?

Below is a copy of the request payload:
```
{"at":1,"cur":["RUB"],"site":{"name":"http://test.localhost:9999","domain":"test.localhost","id":"216200","page":"http://test.localhost:9999/integrationExamples/gpt/hello_world.html?pbjs_debug=true"},"id":"122873215-1617387828-1-46-","user":{"buyerid":0},"device":{"ip":"195.209.111.14","ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36"},"imp":[{"id":"55:test_placement","ext":{"query":"bn=15&custom=111=21ae59f212d93"},"banner":{"w":300,"h":250},"bidfloor":1,"bidfloorcur":"RUB","secure":0,"pmp":{"private_auction":1,"deals":[{"id":"dealidTest","bidfloor":1,"bidfloorcur":"RUB"}]}},{"id":"55:test_placement","ext":{"query":"bn=15&custom=111=21ae59f212d93"},"banner":{"w":300,"h":600},"bidfloor":1,"bidfloorcur":"RUB","secure":0,"pmp":{"private_auction":1,"deals":[{"id":"dealidTest","bidfloor":1,"bidfloorcur":"RUB"}]}}]}
```

Thanks.

File: modules/adriverBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        'domain': win.hostname,
+        'id': siteId,
+        'page': win.href
+      },
+      'id': customID,
+      'user': {
+        'buyerid': 0
+      },
+      'device': {
+        'ip': '195.209.111.14',
```
[the remark this answers is not in the document]
Comment (Fridoom007) [reply modules/adriverBidAdapter.js:46]: Yes, this is required for entry verification.

File: modules/adriverBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          'bidfloor': bidFloor || 0,
+          'bidfloorcur': currency,
+          'secure': 0
+        };
+        if (dealId) {
+          par.pmp = {
+            'private_auction': 1,
+            'deals': [{
+              'id': dealId,
+              'bidfloor': bidFloor || 0,
```
[the remark this answers is not in the document]
Comment (Fridoom007) [reply modules/adriverBidAdapter.js:82]: I added support module "priceFloors" for the bid Adapter, and remove old test "floor" parameter

**Comment by Fridoom007** [comment]
Hi. i made the following changes:
* Add support getFloor()
* Add meta.advertiserDomains
* Update test

Unfortunately ads may not always return, but there is a test page that hosts prebid.js with my adapter. 
http://files.adriver.ru/i.shiryaev/prebid_new/test.html

**Review by jsnellbaker — CHANGES_REQUESTED** [review]
Hi @Fridoom007 

Thank you for making the requested updates and clarifying the other points.

While the changes appear fine, I'm still having trouble trying to get the test bid to come.  I've been refreshing my own and your test page for a while now, but it's never returned any bids.  Is there something else you can try to force a bid to return?

Thanks,
Jason

**Comment by Fridoom007** [comment]
Hi Jason.

We update ad in page: http://files.adriver.ru/i.shiryaev/prebid_new/test.html .
i hope this helps.

Thanks,
 @Fridoom007

**Review by jsnellbaker — APPROVED** [review]
Hi @Fridoom007 

I am still having trouble trying to get an ad on the page, but I had a colleague try and they were able to see it fine.  I think we're fine here now.

**Pull request merged**